aws/resource_aws_instance.go

```diff
@@ -1011,6 +1010,58 @@ func resourceAwsInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	if d.HasChange("root_block_device") {
+		if rbd, ok := d.GetOk("root_block_device"); ok {
+			rbdL := rbd.([]interface{})
+			if len(rbdL) > 1 {
+				return fmt.Errorf("Cannot specify more than one root_block_device.")
+			}
+			for _, r := range rbdL {
+				bd := r.(map[string]interface{})
+				if volumeSize, ok := bd["volume_size"].(int); ok && volumeSize != 0 {
```

Comment:
I think you can simplify a lot of the code above by using `d.HasChange("root_block_device.0.volume_size")` and `d.Get("root_block_device.0.volume_size")` - you can omit the loop, validation, some of the casting etc.